codesign/test/file_codesign_visitor_test.dart

```diff
@@ -246,6 +247,124 @@ void main() {
       expect(messages, contains('Visiting directory ${tempDir.absolute.path}/embedded_zip_${zipFileName.hashCode}'));
     });
 
+    test('procesRemotezip triggers correct workflow', () async {
```

Comment:
What is gsutil fails on download? What if it fails on upload? What happens when notarization fails

Reply:
Sure, added tests for download fail upload succeed, download succeed upload fail, and both download and upload succeed. The notarization related tests were added in previous PRs under the test group('notarization tests: ')